### PR TITLE
Update ALCS Advanced Search

### PR DESCRIPTION
**Pull request by dhaselhan**
* Use new architecture of querying multiple times
* Add stack traces to error logs

### CODE REVIEW

File: services/apps/alcs/src/alcs/search/notification/notification-advanced-search.service.spec.ts [hunk 1]
```diff
@@ -73,63 +74,88 @@ describe('NotificationAdvancedSearchService', () => {
...
-    mockNotificationSubmissionSearchViewRepository.createQueryBuilder.mockReturnValue(
-      mockQuery as any,
+    mockNotificationRepo.find.mockResolvedValue([]);
+    mockNotificationRepo.createQueryBuilder.mockReturnValue(mockQuery);
+    mockNotificationSubmissionRepo.createQueryBuilder.mockReturnValue(
+      mockQuery,
     );
 
-    const result = await service.search(mockSearchDto);
+    const result = await service.search(mockSearchDto, {} as any);
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/search/notification/notification-advanced-search.service.spec.ts:121]: is there a way not to use as any?

File: services/apps/alcs/src/main.ts [hunk 2]
```diff
@@ -116,7 +117,7 @@ function setupLogger() {
   });
 
   const messageFormat = winston.format.printf((info) => {
-    return `${info.timestamp} [${info.level.toUpperCase()}]${info.context ? ` [${info.context}]` : ''} ${info.message}`;
+    return `${info.timestamp} [${info.level.toUpperCase()}]${info.context ? ` [${info.context}]` : ''} ${info.message} ${info[SPLAT] ? `\n${info[SPLAT]}` : ''}`;
```
Comment (mhuseinov) [inline services/apps/alcs/src/main.ts:120]: 👍

File: services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts [hunk 3]
```diff
@@ -1,31 +1,70 @@
...
+
+    let fileNumbers = new Set<string>();
+    if (cachedSearch) {
+      const cachedNumbers = JSON.parse(cachedSearch) as string[];
+      fileNumbers = new Set<string>(cachedNumbers);
+    } else {
+      fileNumbers = await this.searchForFileNumbers(searchDto);
+      await client.setEx(
+        searchKey,
+        180, //Seconds
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts:46]: nitpick: this expiration is shared across all searches would be nice to have it in constant

**Review by mhuseinov — APPROVED** [review]
👍

File: services/apps/alcs/src/alcs/search/notification/notification-advanced-search.service.spec.ts (hunk 1 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/search/notification/notification-advanced-search.service.spec.ts]: Done! Used a mocked queryRunner instead.

File: services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/search/inquiry/inquiry-advanced-search.service.ts]: Done! Moved to search.config.ts but may move again after refactor

**Pull request merged**